main.cpp

```diff
@@ -105,49 +106,52 @@ struct TypeHolder
     }
 
     using FuncPtr = void(*)( Primitive& );
-    TypeHolder& apply( FuncPtr func )
+    Numeric& apply( FuncPtr func )
     {
         if ( func != nullptr )
             func( *a );
         return *this;
     }
 
-    TypeHolder& pow( Primitive rhs )
+    Numeric& pow( Primitive rhs )
     {
         return powInternal ( rhs );
     }
 
-    TypeHolder& operator+=( Primitive rhs )
+    Numeric& operator+=( Primitive rhs )
     {
         *a += rhs;
         return *this;
     }
 
-    TypeHolder& operator-=( Primitive rhs )
+    Numeric& operator-=( Primitive rhs )
     {
         *a -= rhs;
         return *this;
     }
 
-    TypeHolder& operator*=( Primitive rhs )
+    Numeric& operator*=( Primitive rhs )
     {
         *a *= rhs;
         return *this;
     }
 
-    TypeHolder& operator/=( Primitive rhs )
+    Numeric& operator/=( Primitive rhs )
     {   
         if constexpr ( std::is_same<Primitive, int>::value )
         {
-            if ( rhs == 0 )
-            {
-                std::cout << "Can't divide by 0.\n";
-                return *this;
-            }
-            else if ( rhs < std::numeric_limits<Primitive>::epsilon() )
-            {
-                std::cout << "Can't divide by 0.\n";
-                return *this;
+            if( typeid(rhs) == typeid(int) )
```

Comment:
interesting, but please use `if constexpr` like the previous line.  this will cause it to be evaluated at compile time, not at runtime, which is a performance enhancement.
- [ ] revise